### ISSUE TITLE
Add "allowStandalone" option to rule "no-plusplus".

### ISSUE THREAD

**Issue by rkirsling** [eslint/eslint#7134]
The stated purpose of [`no-plusplus`](http://eslint.org/docs/rules/no-plusplus) is to prevent errors caused by ASI/whitespace interaction; however, since such errors are already covered by [`space-unary-ops`](http://eslint.org/docs/rules/space-unary-ops), [`semi`](http://eslint.org/docs/rules/semi), and [`semi-spacing`](http://eslint.org/docs/rules/semi-spacing), this rule really just eliminates syntax from the language, which seems to me a little extreme.

The `allowForLoopAfterthoughts` option is a step in the right direction, but falls just short of addressing a major concern in languages with `++` / `--`: namely, these operators are confusing to reason about **in the presence of other operations**.

As such, I'd like to propose an `allowStandalone` option, which allows `++` / `--` to be used as the sole operation in a statement, instead of demanding that they be rewritten as `+= 1` / `-= 1`.

**Incorrect**

``` js
if (++foo > 3) { ... }

const bar = 10 * foo++;

doStuff(--foo);

return foo--;
```

**Correct**

``` js
++foo;

foo++;

--foo;

foo--;

// ...which should, of course, be taken to include:
foo.bar++;
foo[bar]++;
```

Whether this option should subsume `allowForLoopAfterthoughts` or stand separately is open for discussion. (_I was initially assuming the former, but the latter would likely be easier to use._)

**Comment by platinumazure** [comment]
Are you requesting an option to allow ++/-- on single variables in statement position?

When you talk about replacing +=, that's a bit shaky because assignments are expressions in JavaScript:

```js
doStuff(foo += 1);     // Perfectly legitimate!
```

**Comment by rkirsling** [comment]
Ack, thanks for pointing that out!

I had actually originally written "as standalone statements" and replaced it with "where they can be substituted for `+=` / `-=`", thinking the two to be equivalent, but I'd completely forgotten about the possibility you described.

I'll update the description accordingly.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.

**Comment by rkirsling** [comment]
I'm not sure if I follow what you mean by "consensus"...? There's no expressed disagreement here, only silence (_and a correction to a simple mistake_). My understanding was that the [resources would be my own](http://eslint.org/docs/developer-guide/contributing/rule-changes#implementation-is-your-responsibility) if the rule was deemed acceptable.

I respect the policy to close versus indefinitely defer, but the "evaluating" label seems misleading if the issue is closed without any evidence of evaluation. :confused:

**Comment by kaicataldo** [comment]
Understand the confusion. By "reaching consensus" I mean the process of accepting the issue (which the team has decided requires a champion and three upvotes from the team). Can definitely make that clearer in the future.

We've found that issues that get buried in the backlog without getting accepted end up getting lost (as can be evidenced by the lack of discussion here). Keeping our issues backlog manageable is necessary for keeping the project moving forward, both for us maintainers and also for our contributors.

Feel free to open a new issue with your proposal - chances are much higher it'll get some attention when it's not 6 pages into our issues list!
